synfig-studio/src/gui/modules/mod_palette/dock_paledit.cpp

```diff
@@ -349,6 +369,68 @@ Dock_PalEdit::add_color(const synfig::Color& x)
 	return size()-1;
 }
 
+void
+Dock_PalEdit::add_from_clipboard()
+{
+	if(clipboard_==NULL)
+		clipboard_ = gtk_clipboard_get(GDK_SELECTION_CLIPBOARD);
+	
+	gchar* hexcolor = gtk_clipboard_wait_for_text(clipboard_);
+
+	//Bool to check if the clipboard string is formatted in hexadecimal.  
+	bool hexfrmt = false;
+
+	//Check if clipboard is empty
+	if(hexcolor!=NULL){
```

Comment:
I suggest to extract format check to a separate function. This will make code cleaner.

Reply:
The hex format checking has been moved to a helper function.